### PR TITLE
Obsolete error codes that are not throw by the service

### PR DESCRIPTION
**Pull request by timtay-microsoft**

### CODE REVIEW

File: iothub/service/src/Common/Exceptions/ErrorCode.cs [hunk 1]
```diff
@@ -17,27 +17,35 @@ public enum ErrorCode
...
+        [EditorBrowsable(EditorBrowsableState.Never)]
         DeviceStorageEntitySerializationError = 400007,
 
-        /// <summary>
-        /// An error was encountered validating the storage blob container URI.
-        /// </summary>
-        /// <remarks>
-        /// As this error is in the 4xx HTTP status code range, the service would have detected a problem with the blob container
-        /// URI, such as does not exist, insufficient permissions, or SAS token expiry exceeded.
-        /// </remarks>
```
Comment (timtay-microsoft) [inline iothub/service/src/Common/Exceptions/ErrorCode.cs:33]: No need to document unused error codes

File: iothub/service/src/Common/Exceptions/ErrorCode.cs [hunk 2]
```diff
@@ -106,7 +122,7 @@ public enum ErrorCode
         /// </summary>
         DeviceAlreadyExists = 409001,
 
-        ModuleAlreadyExistsOnDevice = 409301,
+        ModuleAlreadyExistsOnDevice = 409301, // not referenced in service code, but service does throw it?
```
Comment (timtay-microsoft) [inline iothub/service/src/Common/Exceptions/ErrorCode.cs:125]:
```suggestion
        ModuleAlreadyExistsOnDevice = 409301,
```

File: iothub/service/src/Common/Exceptions/ErrorCode.cs [hunk 3]
```diff
@@ -51,30 +59,33 @@ public enum ErrorCode
...
         // Unauthorized - 401
         IotHubNotFound = 401001,
 
-        IotHubUnauthorizedAccess = 401002,
-
         /// <summary>
         /// The SAS token has expired or IoT hub couldn't authenticate the authentication header, rule, or key.
         /// For detailed information, see <see href="https://docs.microsoft.com/azure/iot-hub/iot-hub-troubleshoot-error-401003-iothubunauthorized"/>.
         /// </summary>
-        IotHubUnauthorized = 401003,
```
Comment (timtay-microsoft) [inline iothub/service/src/Common/Exceptions/ErrorCode.cs:65]: git shows this as deleted, but it wasn't. Line 82 on the new file still has it.

File: iothub/service/src/Common/Exceptions/ErrorCode.cs [hunk 4]
```diff
@@ -51,20 +77,36 @@ public enum ErrorCode
...
+        IotHubUnauthorizedAccess = 401002,
+
+        /// <summary>
+        /// Unused error code. Service does not return it and neither does the SDK.
+        /// </summary>
+        [Obsolete("This error does not appear to be thrown by the service.")]
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        IotHubUnauthorized = 401003, // never thrown by service anymore. Replaced by IotHubUnauthorizedAccess
```
Comment (drwill-ms) [inline iothub/service/src/Common/Exceptions/ErrorCode.cs:101]: Want to move the code comment to remarks or something?

Comment (timtay-microsoft) [reply iothub/service/src/Common/Exceptions/ErrorCode.cs:101]: Sure thing

**Review by drwill-ms — APPROVED** [review]

**Pull request merged**